### PR TITLE
qualcommax: ipq60xx: add support for netgear wax214

### PR DESCRIPTION
**Pull request by kirdesde**
Netgear WAX214 is a 802.11 ax dual-band AP
    with PoE. (similar to Engenius EWS357APV3)

    Specifications:

        •     CPU: Qualcomm IPQ6010 Quad core Cortex-A53
        •     RAM: 512MB of DDR3
        •     Storage: 128MB NAND (Macronix MX30UF1G18AC)
        •     Ethernet: 1x 1G RJ45 port (QCA8072) PoE
        •     WIFI:
              2.4GHz: Qualcomm QCN5022 2x2 802.11b/g/n/ax 574 Mbps PHY rate
              5GHz: Qualcomm QCN5052 2x2 802.11a/b/g/n/ac/ax 1201 PHY rate

        •     LEDs:
              4 x GPIO-controlled LEDs
                - 1 Power LED (orange)
                - 1 LAN LED (blue)
                - 1 WIFI 5g LED (blue)
                - 1 WIFI 2g LED (blue)
                black_small_square  Buttons: 1x soft reset
                black_small_square  Power: 12V DC jack or PoE (802.3af )

            An populated serial header is onboard, format is
             1.25mm 4p (DF13A-4P-1.25H)
            RX/TX is working, bootwait is active, secure boot is not
            enabled.

            The root password of the stock firmware is unknown,
            but failsafe mode can be entered to reset the password.

            Installation Instructions:

                - obtain serial access
                - stop auto boot (press "4", Entr boot command line
		  interface)
                - setenv active_fw 0 (to boot from the primary rootfs,
                  or set to 1 to boot from the secondary rootfs
                  partition)
                - saveenv

                - tftpboot the initramfs image
                - bootm

                - copy
		  openwrt-qualcommax-ipq60xx-netgear_wax214-squashfs-factory.ubi
                  to the device
                - write the image to the NAND:
                   - cat /proc/mtd and look for rootfs partition (should be mtd11, or mtd12 if you choose active_fw 1)
                   - ubiformat /dev/mtd11 -f -y openwrt-qualcommax-ipq60xx-netgear_wax214-squashfs-factory.ubi - reboot

            Note: the firmware is senao-based. But I was unable to build
                  a valid senao-header into the image.
                  Maybe they changed the header format and senaoFW isn't
                  working any more.

Pending BDF PR: https://github.com/openwrt/firmware_qca-wireless/pull/32

### CODE REVIEW

File: target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax214.dts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	ethernet-phy-package@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "qcom,qca8075-package";
+		reg = <0>;
+
+		qcom,package-mode = "psgmii";
+
+		qca8072_0: ethernet-phy@4 {
```
Comment (robimarko) [inline target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax214.dts:125]: Would make more sense to be `qca8072_4`, even though its the first PHY in the package

Comment (kirdesde) [reply target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-wax214.dts]: done

**Comment by robimarko** [comment]
Thanks, if CI passes I will merge this.

BTW, any chance you could give the new updated 2.5.0.1 FW a go?
https://github.com/quic/upstream-wifi-fw/tree/main/ath11k-firmware/IPQ6018/hw1.0/2.5.0.1/WLAN.HK.2.5.0.1-03982-QCAHKSWPL_SILICONZ-3

**Comment by kirdesde** [comment]
> Thanks, if CI passes I will merge this.
> 
> BTW, any chance you could give the new updated 2.5.0.1 FW a go? https://github.com/quic/upstream-wifi-fw/tree/main/ath11k-firmware/IPQ6018/hw1.0/2.5.0.1/WLAN.HK.2.5.0.1-03982-QCAHKSWPL_SILICONZ-3

Sure, I'll have a look. Didn't even expect an update for ipq6018

**Comment by kirdesde** [comment]
> > Thanks, if CI passes I will merge this.
> > BTW, any chance you could give the new updated 2.5.0.1 FW a go? https://github.com/quic/upstream-wifi-fw/tree/main/ath11k-firmware/IPQ6018/hw1.0/2.5.0.1/WLAN.HK.2.5.0.1-03982-QCAHKSWPL_SILICONZ-3
> 
> Sure, I'll have a look. Didn't even expect an update for ipq6018

AP-Mode seems to work on both radios.
Had a crash with STA mode, but same occurred with the 2.4.0.1 firmware
Can't check MESH mode

**Comment by robimarko** [comment]
Oh, almost forgot, I merged the BDF but you need to update the ipq-wifi package to point at a newer repo version

**Comment by kirdesde** [comment]
> Oh, almost forgot, I merged the BDF but you need to update the ipq-wifi package to point at a newer repo version

Yeah, I'll open a PR after the device support is merged.

**Comment by robimarko** [comment]
> > Oh, almost forgot, I merged the BDF but you need to update the ipq-wifi package to point at a newer repo version
> 
> Yeah, I'll open a PR after the device support is merged.

No need, just add the bump before the device commit here, otherwise the support for this board will be technically broken

**Comment by kirdesde** [comment]
> > > > Oh, almost forgot, I merged the BDF but you need to update the ipq-wifi package to point at a newer repo version
> > > 
> > > 
> > > Yeah, I'll open a PR after the device support is merged.
> > 
> > 
> > No need, just add the bump before the device commit here, otherwise the support for this board will be technically broken
> 
> done

Should I squash that into one commit? Otherwise test formalities failing

**Comment by kirdesde** [comment]
> No, you just need to a add a commit description to the bump commit.
> 
> Also, put it before the device addition commit.

Done

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**